# Obtención de datos geográficos — INEGI

## Proyecto Agronomía

Esta libreta documenta la obtención y preparación de la información
geográfica utilizada en el proyecto.

Las geometrías de las entidades federativas y municipios de México
se obtienen mediante los servicios geográficos de INEGI.

La lógica de adquisición y procesamiento se encuentra implementada
en el paquete `proyecto_agronomia`, por lo que esta libreta funciona
como una interfaz reproducible para ejecutar, inspeccionar y validar
el proceso.

## Objetivo

Obtener:

- geometrías de los 32 estados de México;
- geometrías municipales;
- coordenadas representativas de cada municipio.

Estas coordenadas serán utilizadas posteriormente para consultar
las variables climáticas históricas.

## Flujo de trabajo

1. Descargar o cargar las geometrías municipales de INEGI.
2. Descargar o cargar las geometrías estatales.
3. Generar un punto representativo para cada municipio.
4. Validar claves geográficas y coordenadas.
5. Guardar el resultado para la etapa climática.

In [1]:
from proyecto_agronomia.dataset import (
    obtener_estados_inegi,
    obtener_municipios_inegi,
)

from proyecto_agronomia.features import (
    guardar_coordenadas_municipios,
)

2026-10-07 07:53:49.290 | INFO     | proyecto_agronomia.config:<module>:14 - PROJ_ROOT path is: /home/david/Documents/Repositories/academica/proyecto-agronomia-main


## 1. Geometrías municipales

Se obtienen las geometrías de los municipios de México.

La función reutiliza el archivo local cuando este ya existe, evitando
realizar nuevamente la descarga.

In [2]:
municipios_geo = obtener_municipios_inegi()

municipios_geo.head()

Municipios INEGI:   0%|          | 0/32 [00:00<?, ?it/s]

2026-10-07 07:53:53.548 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 01.


Municipios INEGI:   3%|▎         | 1/32 [00:03<01:52,  3.64s/it]

2026-10-07 07:53:57.190 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 02.


Municipios INEGI:   6%|▋         | 2/32 [00:08<02:12,  4.41s/it]

2026-10-07 07:54:02.138 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 03.


Municipios INEGI:   9%|▉         | 3/32 [00:12<01:59,  4.11s/it]

2026-10-07 07:54:05.891 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 04.


Municipios INEGI:  12%|█▎        | 4/32 [00:13<01:21,  2.93s/it]

2026-10-07 07:54:07.001 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 05.


Municipios INEGI:  16%|█▌        | 5/32 [00:15<01:06,  2.46s/it]

2026-10-07 07:54:08.642 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 06.


Municipios INEGI:  19%|█▉        | 6/32 [00:16<00:55,  2.12s/it]

2026-10-07 07:54:10.085 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 07.


Municipios INEGI:  22%|██▏       | 7/32 [00:21<01:19,  3.18s/it]

2026-10-07 07:54:15.466 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 08.


Municipios INEGI:  25%|██▌       | 8/32 [00:28<01:42,  4.28s/it]

2026-10-07 07:54:22.094 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 09.


Municipios INEGI:  28%|██▊       | 9/32 [00:30<01:21,  3.52s/it]

2026-10-07 07:54:23.959 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 10.


Municipios INEGI:  31%|███▏      | 10/32 [00:35<01:30,  4.12s/it]

2026-10-07 07:54:29.401 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 11.


Municipios INEGI:  34%|███▍      | 11/32 [00:39<01:24,  4.04s/it]

2026-10-07 07:54:33.281 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 12.


Municipios INEGI:  38%|███▊      | 12/32 [00:45<01:31,  4.59s/it]

2026-10-07 07:54:39.109 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 13.


Municipios INEGI:  41%|████      | 13/32 [00:50<01:28,  4.68s/it]

2026-10-07 07:54:44.015 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 14.


Municipios INEGI:  44%|████▍     | 14/32 [00:58<01:40,  5.57s/it]

2026-10-07 07:54:51.633 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 15.


Municipios INEGI:  47%|████▋     | 15/32 [01:07<01:51,  6.59s/it]

2026-10-07 07:55:00.581 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 16.


Municipios INEGI:  50%|█████     | 16/32 [01:12<01:38,  6.17s/it]

2026-10-07 07:55:05.785 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 17.


Municipios INEGI:  53%|█████▎    | 17/32 [01:14<01:12,  4.86s/it]

2026-10-07 07:55:07.595 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 18.


Municipios INEGI:  56%|█████▋    | 18/32 [01:16<00:58,  4.21s/it]

2026-10-07 07:55:10.296 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 19.


Municipios INEGI:  59%|█████▉    | 19/32 [01:19<00:49,  3.79s/it]

2026-10-07 07:55:13.107 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 20.


Municipios INEGI:  62%|██████▎   | 20/32 [01:28<01:03,  5.27s/it]

2026-10-07 07:55:21.830 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 21.


Municipios INEGI:  66%|██████▌   | 21/32 [01:38<01:13,  6.64s/it]

2026-10-07 07:55:31.658 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 22.


Municipios INEGI:  69%|██████▉   | 22/32 [01:40<00:55,  5.51s/it]

2026-10-07 07:55:34.539 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 23.


Municipios INEGI:  72%|███████▏  | 23/32 [01:45<00:45,  5.06s/it]

2026-10-07 07:55:38.551 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 24.


Municipios INEGI:  75%|███████▌  | 24/32 [01:48<00:35,  4.46s/it]

2026-10-07 07:55:41.605 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 25.


Municipios INEGI:  78%|███████▊  | 25/32 [01:52<00:30,  4.43s/it]

2026-10-07 07:55:45.953 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 26.


Municipios INEGI:  81%|████████▏ | 26/32 [01:54<00:21,  3.61s/it]

2026-10-07 07:55:47.670 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 27.


Municipios INEGI:  84%|████████▍ | 27/32 [01:56<00:16,  3.32s/it]

2026-10-07 07:55:50.309 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 28.


Municipios INEGI:  88%|████████▊ | 28/32 [01:57<00:10,  2.69s/it]

2026-10-07 07:55:51.534 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 29.


Municipios INEGI:  91%|█████████ | 29/32 [02:01<00:08,  2.92s/it]

2026-10-07 07:55:54.975 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 30.


Municipios INEGI:  94%|█████████▍| 30/32 [02:14<00:11,  5.92s/it]

2026-10-07 07:56:07.889 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 31.


Municipios INEGI:  97%|█████████▋| 31/32 [02:16<00:04,  4.67s/it]

2026-10-07 07:56:09.645 | INFO     | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:45 - Descargando municipios del estado 32.


Municipios INEGI: 100%|██████████| 32/32 [02:21<00:00,  4.42s/it]


2026-10-07 07:56:29.815 | SUCCESS  | proyecto_agronomia.datasets.inegi:obtener_municipios_inegi:78 - Municipios INEGI guardados: 2,478


,geometry,cvegeo,cve_ent,cve_mun,nomgeo,cve_cab,nom_cab,pob_total,pob_femenina,pob_masculina,total_viviendas_habitadas
0,"MULTIPOLYGON (((-102.10732 22.07476, -102.1069...",01001,01,001,Aguascalientes,0001,Aguascalientes,948990,486917,462073,266942
1,"MULTIPOLYGON (((-102.45537 22.31212, -102.4550...",01008,01,008,San José de Gracia,0001,NaN,9552,5020,4532,2381
2,"MULTIPOLYGON (((-102.15938 22.099, -102.15638 ...",01011,01,011,San Francisco de los Romo,0001,NaN,61997,31292,30705,16391
3,"MULTIPOLYGON (((-102.17738 22.36243, -102.1796...",01009,01,009,Tepezalá,0001,NaN,22485,11371,11114,5425
4,"MULTIPOLYGON (((-102.22684 22.37393, -102.2266...",01007,01,007,Rincón de Romos,0001,NaN,57369,29268,28101,13584


In [3]:
print(f"Municipios disponibles: {len(municipios_geo):,}")
print(f"CRS: {municipios_geo.crs}")
print(f"Columnas: {municipios_geo.columns.tolist()}")

Municipios disponibles: 2,478
CRS: EPSG:4326
Columnas: ['geometry', 'cvegeo', 'cve_ent', 'cve_mun', 'nomgeo', 'cve_cab', 'nom_cab', 'pob_total', 'pob_femenina', 'pob_masculina', 'total_viviendas_habitadas']


## 2. Geometrías estatales

Se obtienen las geometrías correspondientes a las 32 entidades
federativas de México.

In [4]:
estados_geo = obtener_estados_inegi()

estados_geo.head()

Estados INEGI:   0%|          | 0/32 [00:00<?, ?it/s]

2026-10-07 07:56:34.188 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 01.


Estados INEGI:   3%|▎         | 1/32 [00:00<00:19,  1.60it/s]

2026-10-07 07:56:34.814 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 02.


Estados INEGI:   6%|▋         | 2/32 [00:06<01:43,  3.45s/it]

2026-10-07 07:56:40.250 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 03.


Estados INEGI:   9%|▉         | 3/32 [00:14<02:44,  5.68s/it]

2026-10-07 07:56:48.573 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 04.


Estados INEGI:  12%|█▎        | 4/32 [00:19<02:32,  5.46s/it]

2026-10-07 07:56:53.692 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 05.


Estados INEGI:  16%|█▌        | 5/32 [00:20<01:47,  3.98s/it]

2026-10-07 07:56:55.056 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 06.


Estados INEGI:  19%|█▉        | 6/32 [00:21<01:15,  2.91s/it]

2026-10-07 07:56:55.888 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 07.


Estados INEGI:  22%|██▏       | 7/32 [00:24<01:09,  2.76s/it]

2026-10-07 07:56:58.343 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 08.


Estados INEGI:  25%|██▌       | 8/32 [00:24<00:49,  2.06s/it]

2026-10-07 07:56:58.898 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 09.


Estados INEGI:  28%|██▊       | 9/32 [00:25<00:36,  1.57s/it]

2026-10-07 07:56:59.382 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 10.


Estados INEGI:  31%|███▏      | 10/32 [00:26<00:34,  1.55s/it]

2026-10-07 07:57:00.910 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 11.


Estados INEGI:  34%|███▍      | 11/32 [00:27<00:25,  1.24s/it]

2026-10-07 07:57:01.421 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 12.


Estados INEGI:  38%|███▊      | 12/32 [00:28<00:22,  1.10s/it]

2026-10-07 07:57:02.226 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 13.


Estados INEGI:  41%|████      | 13/32 [00:28<00:19,  1.03s/it]

2026-10-07 07:57:03.087 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 14.


Estados INEGI:  44%|████▍     | 14/32 [00:31<00:29,  1.62s/it]

2026-10-07 07:57:06.066 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 15.


Estados INEGI:  47%|████▋     | 15/32 [00:34<00:32,  1.93s/it]

2026-10-07 07:57:08.720 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 16.


Estados INEGI:  50%|█████     | 16/32 [00:36<00:31,  1.95s/it]

2026-10-07 07:57:10.712 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 17.


Estados INEGI:  53%|█████▎    | 17/32 [00:36<00:22,  1.50s/it]

2026-10-07 07:57:11.172 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 18.


Estados INEGI:  56%|█████▋    | 18/32 [00:38<00:21,  1.52s/it]

2026-10-07 07:57:12.737 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 19.


Estados INEGI:  59%|█████▉    | 19/32 [00:40<00:20,  1.59s/it]

2026-10-07 07:57:14.491 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 20.


Estados INEGI:  62%|██████▎   | 20/32 [00:44<00:26,  2.23s/it]

2026-10-07 07:57:18.199 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 21.


Estados INEGI:  66%|██████▌   | 21/32 [00:45<00:23,  2.13s/it]

2026-10-07 07:57:20.095 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 22.


Estados INEGI:  69%|██████▉   | 22/32 [00:46<00:16,  1.69s/it]

2026-10-07 07:57:20.774 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 23.


Estados INEGI:  72%|███████▏  | 23/32 [00:48<00:15,  1.77s/it]

2026-10-07 07:57:22.716 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 24.


Estados INEGI:  75%|███████▌  | 24/32 [00:49<00:12,  1.56s/it]

2026-10-07 07:57:23.780 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 25.


Estados INEGI:  78%|███████▊  | 25/32 [00:50<00:09,  1.29s/it]

2026-10-07 07:57:24.434 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 26.


Estados INEGI:  81%|████████▏ | 26/32 [00:55<00:14,  2.37s/it]

2026-10-07 07:57:29.321 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 27.


Estados INEGI:  84%|████████▍ | 27/32 [00:55<00:09,  1.90s/it]

2026-10-07 07:57:30.136 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 28.


Estados INEGI:  88%|████████▊ | 28/32 [00:58<00:07,  1.99s/it]

2026-10-07 07:57:32.322 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 29.


Estados INEGI:  91%|█████████ | 29/32 [00:58<00:04,  1.58s/it]

2026-10-07 07:57:32.968 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 30.


Estados INEGI:  94%|█████████▍| 30/32 [01:00<00:03,  1.51s/it]

2026-10-07 07:57:34.311 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 31.


Estados INEGI:  97%|█████████▋| 31/32 [01:00<00:01,  1.31s/it]

2026-10-07 07:57:35.148 | INFO     | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:114 - Descargando estado 32.


Estados INEGI: 100%|██████████| 32/32 [01:01<00:00,  1.93s/it]


2026-10-07 07:57:40.206 | SUCCESS  | proyecto_agronomia.datasets.inegi:obtener_estados_inegi:144 - Entidades geográficas guardadas: 32


,geometry,cvegeo,cve_ent,nomgeo,nom_abrev,pob_total,pob_femenina,pob_masculina,total_viviendas_habitadas
0,"MULTIPOLYGON (((-102.29739 22.45527, -102.2955...",01,01,Aguascalientes,Ags.,1425607,728924,696683,386671
1,"MULTIPOLYGON (((-114.11561 28.09639, -114.1152...",02,02,Baja California,BC,3769020,1868431,1900589,1149563
2,"MULTIPOLYGON (((-109.89451 22.87476, -109.8946...",03,03,Baja California Sur,BCS,798447,392568,405879,240660
3,"MULTIPOLYGON (((-91.9565 20.197, -91.95636 20....",04,04,Campeche,Camp.,928363,471424,456939,260824
4,"MULTIPOLYGON (((-102.31549 29.88, -102.31509 2...",05,05,Coahuila de Zaragoza,Coah.,3146771,1583102,1563669,901249


In [5]:
print(f"Entidades federativas: {len(estados_geo):,}")
print(f"CRS: {estados_geo.crs}")

Entidades federativas: 32
CRS: EPSG:4326


## 3. Coordenadas representativas municipales

Para consultar posteriormente información climática es necesario
asociar cada municipio con una coordenada.

Se calcula un punto representativo dentro de cada geometría municipal
y posteriormente se transforma al sistema de coordenadas geográficas
EPSG:4326.

In [5]:
municipios_coordenadas = guardar_coordenadas_municipios(
    municipios_geo
)

municipios_coordenadas.head()

2026-10-07 08:04:16.560 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:303 - Municipios: 2,478
2026-10-07 08:04:16.561 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:306 - Claves duplicadas: 0
2026-10-07 08:04:16.561 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:309 - Latitudes faltantes: 0
2026-10-07 08:04:16.561 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:312 - Longitudes faltantes: 0
2026-10-07 08:04:16.561 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:316 - Rango latitud: 14.634916 → 32.416923
2026-10-07 08:04:16.562 | INFO     | proyecto_agronomia.features:guardar_coordenadas_municipios:322 - Rango longitud: -116.882750 → -86.914552
2026-10-07 08:04:16.578 | SUCCESS  | proyecto_agronomia.features:guardar_coordenadas_municipios:339 - Coordenadas municipales guardadas en /home/david/Documents/Repositories/academica/proyecto-agronomia-main/data/interim/ine

,Idestado,Idmunicipio,cvegeo,Nommunicipio,Latitud,Longitud
0,1,1,01001,Aguascalientes,21.848487,-102.305449
1,1,8,01008,San José de Gracia,22.154297,-102.524108
2,1,11,01011,San Francisco de los Romo,22.021698,-102.251799
3,1,9,01009,Tepezalá,22.243608,-102.192713
4,1,7,01007,Rincón de Romos,22.250711,-102.348371


In [7]:
print(
    f"Municipios: {len(municipios_coordenadas):,}"
)

print(
    "Claves duplicadas:",
    municipios_coordenadas.duplicated(
        ["Idestado", "Idmunicipio"]
    ).sum(),
)

print(
    "Coordenadas faltantes:",
    municipios_coordenadas[
        ["Latitud", "Longitud"]
    ].isna().sum().sum(),
)

print(
    f"Latitud: "
    f"{municipios_coordenadas['Latitud'].min():.4f} → "
    f"{municipios_coordenadas['Latitud'].max():.4f}"
)

print(
    f"Longitud: "
    f"{municipios_coordenadas['Longitud'].min():.4f} → "
    f"{municipios_coordenadas['Longitud'].max():.4f}"
)

Municipios: 2,478
Claves duplicadas: 0
Coordenadas faltantes: 0
Latitud: 14.6349 → 32.4169
Longitud: -116.8828 → -86.9146


## Resultado

La información geográfica de INEGI quedó preparada para las siguientes
etapas del proyecto.

El conjunto de coordenadas municipales utiliza como identificadores:

- `Idestado`
- `Idmunicipio`

Estas claves permiten relacionar posteriormente la información
geográfica con los registros agrícolas.

Las coordenadas `Latitud` y `Longitud` se utilizarán para consultar
los datos climáticos históricos de cada municipio.